src/kudu/consensus/consensus_queue.cc

```diff
@@ -1349,9 +1349,16 @@ void PeerMessageQueue::EndWatchForSuccessor() {
 }
 
 Status PeerMessageQueue::GetNextRoutingHopFromLeader(const string& dest_uuid, string* next_hop) const {
```

Comment:
Is the extra "FromLeader" required in this function name? It seems to me more confusing as I can't understand how the Leader is coming into play here.

Reply:
"PeerMessageQueue::GetNextRoutingHopFromLeader()" is only called on the leader (to get the next hop along the routing tree). Hence the name (suggesting next hop to go from the leader). I can change it if it no longer makes sense.